keras_retinanet/bin/train.py

```diff
@@ -275,6 +275,8 @@ def csv_list(string):
     group.add_argument('--weights',           help='Initialize the model with weights from a file.')
     group.add_argument('--no-weights',        help='Don\'t initialize the model with any weights.', dest='imagenet_weights', action='store_const', const=False)
 
+    parser.add_argument('--backbone',        help='Backbone model used by retinanet.', default='resnet', type=str)
+    parser.add_argument('--backbone_param',  help='Backbone model parameter.', default='50', type=int)
```

Comment:
Prefer to use `-` instead of `_` (so `--backbone-param`).

Also, @de-vri-es , can you give your opinion on this part? I know you have some thoughts about it.

Reply:
Personally, I think this should be a singular parameter. We refer to models by ResNet50, ResNet101 etc. And the `param` number may not hold for different backbones. It's far better to use singular names for things, rather than guessing that all conceivable models are `name + depth` based.

Reply:
I don't really like splitting the backbone definition into two parameters. For example: I'd prefer `--backbone resnet50` over `--backbone resnet --backbone-param 50`. 

That also helps catching errors in the command line options quickly: `resnet42` shouldn't be accepted. `parse_args` and/or `check_args` should fail on that.

Reply:
You are right. I'm fixing this and in the same time doing a small refactoring. Eager to hear your opinion.